trpcultivate_phenotypes/src/Service/TripalCultivatePhenotypesTraitsService.php

```diff
@@ -414,4 +418,97 @@ public function getMethodUnitDataType($unit_id) {
 
     return $data_type ?? 0;
   }
-}
+
+  /**
+   * Get trait, method and unit combination.
+   * 
+   * @param string|int $trait
+   *   A string value is the trait name, whereas an integer value is the trait id number.
+   * @param string|int $method
+   *   A string value is the trait method short name, whereas an integer value is the method id number.
+   * @param string|int $unit
+   *   A string value is the method unit name, whereas an integer value is the unit id number.
+   * 
+   * @return array
+   *   An associative array where the keys are trait, method and unit and the values
+   *   are the cvterm records for each key.
+   * 
+   *   0 if any one of trait, method or unit did not return any record.
+   * 
+   * @dependencies
+   *   getTrait(), getTraitMethod(), getMethodUnit() and getMethodUnitDataType().
+   */
+  public function getTraitMethodUnitCombo(string|int $trait, string|int $method, string|int $unit) {
+    // Make sure that each parameter has a valid value.
+    foreach([$trait, $method, $unit] as $param) {
+      if (empty($param) || $param <= 0) {
+        return 0;
```

Comment:
Anywhere there is an error reached in an API, an exception with a meaningful message should be thrown. I'm seeing a lot of `return 0;` throughout these getters and it makes debugging very difficult.
```suggestion
        throw \Exception('The trait, method and unit passed into getTraitMethodUnitCombo() is required; however, one of them was empty.');
```

Reply:
This one makes sense. Good catch!

Reply:
🥰 Please update all the other `return 0;` instances in the getter functions 🙂